File: source_code/eightqueen/eightqueen.py
class eightqueen:
    def __init__(self, *args, **kwargs):
        if len(args)==2:
            self.board = list(args[0])
            self.lastqueen = args[1]
        else:
            self.board = list([0, 0, 0, 0, 0, 0, 0, 0])
            self.lastqueen = 0
    
    def __str__(self):
        return str(self.board)

    def __repr__(self):
        return str(self)

    def isgoal(self):
        if self.lastqueen == 8:
            return True
        return False

    def successors(self):
        return []

File: source_code/eightqueen/eightqueen.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"successor.h"
#include"eightqueen.h"

eightqueen *new_empty_board() {
  eightqueen *e = (eightqueen *)malloc(sizeof(eightqueen));
  memset(e->board, 0, sizeof(int)*8);
  e->lastqueen = 0;
  return e;
}

eightqueen *new_board(int *b, int l) {
  eightqueen *e = (eightqueen *)malloc(sizeof(eightqueen));
  memcpy(e->board, b, sizeof(int)*8);
  e->lastqueen = l;
  return e;
}

int is_goal(eightqueen *e) {
  if (e->lastqueen == 8)
    return 1;
  return 0;
}

void successors(eightqueen *e, successor **list) {
  int idx = 0;
  
  *list = (successor *)calloc(8, sizeof(successor));
}

void print_board(eightqueen *e) {
  int i;
  printf("[");
  for(i=0; i<7; i++)
    printf("%d, ", e->board[i]);
  printf("%d]\n", e->board[7]);
}
